### DISCUSSION TITLE
What's the difference between exec, execO, and execA?

### DISCUSSION THREAD

**Discussion by AlexErrant** [vlcn-io/cr-sqlite#83] [Q&A] [answered]
And when should I use one over another?

https://github.com/vlcn-io/cr-sqlite/blob/main/js/wasm-esm/wa-crsqlite/src/wrapper.ts#L135-L158

Obviously something to do with [`computeCacheKey`](https://github.com/vlcn-io/cr-sqlite/blob/main/js/wasm-esm/wa-crsqlite/src/wrapper.ts#L100-L123), but I've no idea what that does 😅

**Comment by tantaman** [comment, marked as answer]
`execO` returns an Object `{col1: val1, col2: val2, ...}` for each row

`execA` returns an Array: `[col1val, col2val, ...]` for each row

I'm open to better ideas for the name :) it is pretty opaque.

**Reply by AlexErrant** [reply]
D'oh! That's what I get for trying to do static analysis instead of, y'know... running the code.
